optimade/filtertransformers/mongo.py

```diff
@@ -476,6 +476,36 @@ def replace_str_id_with_objectid(subdict, prop, expr):
             filter_, check_for_id_key, replace_str_id_with_objectid
         )
 
+    def _apply_mongo_date_filter(self, filter_: dict) -> dict:
+        """This method loops through the query and replaces any operations
+        on suspected timestamp properties with the corresponding operation
+        on a BSON `DateTime` type.
+        """
+
+        def check_for_timestamp_field(prop, _):
+            """ Find cases where the query dict is operating on a timestamp field. """
+            if self.mapper is not None:
+                prop = self.mapper.alias_of(prop)
+            return prop == "last_modified"
```

Comment:
Just a heads up, eventually this can use the mapper/schema mechanism to find all implementation-defined fields of type `DataType.TIMESTAMP`, but that will have to wait